### ISSUE TITLE
JDK-8193445 JavaFX CSS is applied redundantly leading to significant performance degradation

### ISSUE THREAD

**Issue by buedi** [javafxports/openjdk-jfx#412]
Is there any workaround to prevent this JavaFX performance issue?
I had massive performance problems and was forced to switch back our application to Java 8u171

[https://bugs.openjdk.java.net/browse/JDK-8193445](https://bugs.openjdk.java.net/browse/JDK-8193445)

Best regards 
Daniel B.

Example:
```
package perfo; 

import javafx.application.Application; 
import javafx.geometry.Insets; 
import javafx.scene.Scene; 
import javafx.scene.control.TextField; 
import javafx.scene.control.ToolBar; 
import javafx.scene.layout.BorderPane; 
import javafx.scene.layout.HBox; 
import javafx.scene.text.Text; 
import javafx.stage.Stage; 

/* 
 * To change this license header, choose License Headers in Project Properties. 
 * To change this template file, choose Tools | Templates 
 * and open the template in the editor. 
 */ 
public class Perfo extends Application { 

    private BorderPane pane; 

    @Override 
    public void start(Stage primaryStage) throws Exception { 
        pane = new BorderPane(); 
        TextField textField = new TextField("400"); 
        pane.setTop(new ToolBar(textField)); 
        textField.setOnAction(e -> { 
            HBox hbox = new HBox(); 
            int nb = Integer.parseInt(textField.getText()); 
            for (int i = 0; i < nb; i++) { 

                hbox = new HBox(new Text("y"), hbox); 
                final HBox h = hbox; 
                h.setPadding(new Insets(1)); 
            } 
            pane.setCenter(hbox); 

        }); 

        Scene scene = new Scene(pane); 
        primaryStage.setScene(scene); 
        primaryStage.sizeToScene(); 
        primaryStage.show(); 
    } 

    public static void main(String[] args) { 
        launch(args); 
    } 
} 
```

**Comment by DeanWookey** [comment]
You can try the suggestion in this thread: https://mail.openjdk.java.net/pipermail/openjfx-dev/2019-January/023083.html

As mentioned there, I haven't actually tried the code.

**Comment by kevinrushforth** [comment]
This GitHub issue tracker is not actively tracked or managed. I note that the bug in question has already been fixed by PR openjdk/jfx#34
